src/cct.rs

```diff
@@ -344,39 +344,68 @@ fn test_cct(){
+        return None;
+    };
+
+    // calculate the cct to test.
+    let cct: CCT = xyz0.try_into().unwrap();
+
+    // calculate XYZ, should not fail, a CCT and Duv very close to already fetted values.
+    let xyz: XYZ = cct.try_into().unwrap();
+    let d_uv = xyz.uv_prime_distance(&xyz0);
+    Some(d_uv)
+}
 
-    // Test round trip random values, and check the difference by distance in uv-prime space.  For a
-    // 4096 size table, distances are found to be less than 6E-5 over the full range of temperatures
-    // (1000K, 1_000_000K) and duv values (-0.05, 0.05). This is a relatively slow test, as it tends
-    // to fill the Robertson lookup table fully, with each entry requiring to calculate tristimulus
-    // values from a Planckian spectrum. It will speed up when more than ~ 5_000 values are tested, 
-    // as the table will be completely calculated.
-
-    let seed: u64 = 27; // using a fixed seed, generating the same random numbers at every run.
-    let mut rng = <rand::rngs::StdRng as rand::SeedableRng>::seed_from_u64(seed);
-
-    for i in 0..100 {
-      //  let mut rng = rand::thread_rng();
-        let mired = rand::Rng::gen_range(&mut rng, 1.0..1000.0); // mired temp
-        let t = 1E6/mired;
-        let d = rand::Rng::gen_range(&mut rng, -0.05..0.05);
-        
-        // Get a CCT to test. Unwrap OK as range restricted above.
-        let cct0  = CCT::try_new(t,d).unwrap();
-
-        // calculate XYZ0, skip value if not a valid point
-        let Ok(xyz0):Result<XYZ,_> = CCT::try_new(t,d).unwrap().try_into() else { continue;};
-        
-        // calculate the cct to test.
-        let cct: CCT = xyz0.try_into().unwrap();
-
-        // calculate XYZ, should not fail, a CCT and Duv very close to already fetted values.
-        let xyz: XYZ = cct.try_into().unwrap();
-        let d = xyz.uv_prime_distance(&xyz0);
-        assert_ulps_eq!(d, 0.0, epsilon=6E-5);
+// Test round trip random values, and check the difference by distance in uv-prime space.  For a
+// 4096 size table, distances are found to be less than 5.8E-5 over the full range of temperatures
+// (1000K, 1_000_000K) and duv values (-0.05, 0.05). This is a relatively slow test, as it tends
+// to fill the Robertson lookup table fully, with each entry requiring to calculate tristimulus
+// values from a Planckian spectrum. It will speed up when more than ~ 5_000 values are tested,
+// as the table will be completely calculated.
+#[test]
+fn test_cct_exhaustive() {
+    const MIRED_START: f64 = 1.0;
+    const MIRED_END: f64 = 1000.0;
+    const MIRED_STEP: f64 = 0.05;
+
+    const D_START: f64 = -0.05;
+    const D_END: f64 = 0.05;
+    const D_STEP: f64 = 0.005;
+
+    let mut mired = MIRED_START;
+    while mired < MIRED_END {
+        // Make the d range exclusive in both ends, since it yields errors at the extremes
+        let mut d = D_START + D_STEP;
+        while d <= D_END - D_STEP {
```

Comment:
Is this to be expected? For some values of `mired` running the test with `d = -0.05` or `d = 0.05` yields the error `CCTDuvHighError` or `CCTDuvLowError`. You can easily test this by removing the `+ D_STEP` and `- D_STEP` part of this code and run it locally.

Reply:
I think this is a good finding uncovered from rewriting the tests this way. The previous code:
```rust
let d = rand::Rng::gen_range(&mut rng, -0.05..0.05);
```
Indicates that the test should pass for any value of `d` from `-0.05` inclusive to `0.05` exclusive. But this is apparently not the case. Too close to either end of the range and an error in created instead.

Reply:
I pushed a new commit where I make the code around this a bit cleaner IMO. I made the range inclusive again, but changed the `D_START` and `D_END` values to skip the extremes instead. And improved the comment around it. I hop the comment is correct. I did not dig into too much detail around why this happens.